src/metabase/models/card.clj

```diff
@@ -400,11 +400,62 @@
                         {:status-code 400})))))
   nil)
 
+(defn- dashboard-internal-card? [card]
+  (boolean (:dashboard_id card)))
+
+(def ^:dynamic ^:private *updating-dashboard* false)
+
+(defmacro with-allowed-changes-to-dashboard-internal-card
```

Comment:
This (and other functions in this namespace) have a similar issue to what I mentioned above. The phrase "dashboard-internal" is hard to make sense of when it's in the middle of a larger kebab-case function name. It's fine when you're already familiar with the concept but hard to parse with fresh eyes.

Not sure what the best solution is. Maybe just flip it around as `internal dashboard card`?

Reply:
I think flipping it around makes sense. It's still not totally clear, but seems a bit clearer, so seems good for now. Maybe in the long run we can try to figure out a better name for "Dashboard Questions", "Internal Dashboard Cards", "Dashboard Internal Cards", etc. Feels like right now we have lots of different names, all of which are a bit confusing and some of which have possibly overloaded meanings (e.g. "Dashboard Questions" feels right to me, but seems easy to confuse with DashboardCards).